suite-native/transactions/src/components/TransactionDetail/TransactionDetailAddressesSection.tsx

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import { TouchableOpacity } from 'react-native-gesture-handler';
+
+import { Box, Text } from '@suite-native/atoms';
+import { AccountAddressFormatter } from '@suite-native/formatters';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { CryptoIcon, CryptoIconName } from '@trezor/icons';
+
+import { SummaryRow } from './TransactionDetailStatusSection';
+
+type TransactionDetailAddressesSectionProps = {
+    addresses: string[];
+    addressesType: 'inputs' | 'outputs';
+    onShowMore: () => void;
+    cryptoIcon?: CryptoIconName;
+};
+
+const showMoreButtonStyle = prepareNativeStyle(_ => ({ flexDirection: 'row' }));
+
+const hiddenTransactionsCountStyle = prepareNativeStyle(utils => ({
+    justifyContent: 'center',
+    marginLeft: utils.spacings.small,
+    backgroundColor: utils.colors.backgroundNeutralSubtleOnElevation1,
+    borderRadius: utils.borders.radii.round,
+    paddingHorizontal: utils.spacings.small,
+    paddingVertical: utils.spacings.small / 4,
+}));
+
+const addressTextStyle = prepareNativeStyle(() => ({
+    maxWidth: 160,
+}));
+
+const stepperDotWrapperStyle = prepareNativeStyle(utils => ({
+    justifyContent: 'center',
+    alignItems: 'center',
+    marginTop: utils.spacings.small,
+    backgroundColor: utils.colors.backgroundSurfaceElevation2,
+    width: utils.spacings.medium,
+    height: utils.spacings.medium,
+    borderRadius: utils.borders.radii.round,
+}));
+
+const stepperDotStyle = prepareNativeStyle(utils => ({
+    width: utils.spacings.small / 2,
+    height: utils.spacings.small / 2,
+    borderRadius: utils.borders.radii.round,
+    backgroundColor: utils.colors.backgroundNeutralSubdued,
+}));
+
+const coinIconWrapperStyle = prepareNativeStyle(utils => ({
+    alignSelf: 'flex-start',
+    padding: utils.spacings.small * 1.5,
+    backgroundColor: utils.colors.backgroundSurfaceElevation2,
+    borderRadius: utils.borders.radii.round,
+}));
+
+const TransactionDetailSummaryStepper = () => {
+    const { applyStyle } = useNativeStyles();
+
+    return (
+        <Box style={applyStyle(stepperDotWrapperStyle)}>
+            <Box style={applyStyle(stepperDotStyle)} />
+        </Box>
+    );
+};
+
+export const TransactionDetailAddressesSection = ({
+    addressesType,
+    addresses,
+    onShowMore,
+    cryptoIcon,
+}: TransactionDetailAddressesSectionProps) => {
+    const { applyStyle } = useNativeStyles();
+
+    const titlePrefix = addressesType === 'inputs' ? 'From' : 'To';
+    const formattedTitle =
+        addresses.length > 1 ? `${titlePrefix}  ·  ${addresses.length}` : titlePrefix;
+
+    const displayedAddresses = addresses.slice(0, 2);
+    const isShowMoreButtonVisible = addresses.length > 2;
+    const hiddenAddressesCount = addresses.length - 2;
```

Comment:
Did you consider using selectors for these?

Reply:
I don't think a selector is needed here. It is the only place where it would be called and those are elementary operations so memorization would be useless.